peekingduck/pipeline/nodes/model/osnetv1/osnet_files/osnet_ain.py

```diff
@@ -0,0 +1,318 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+OSNet-AIN model architecture.
+"""
+
+from __future__ import division, absolute_import
+from typing import Any, List, Optional
+import torch
+from torch import nn
+from torch.nn import functional as F
+from peekingduck.pipeline.nodes.model.osnetv1.osnet_files.network_blocks import (
+    ConvLayer,
+    Conv1x1,
+    Conv1x1Linear,
+    LightConv3x3,
+    ChannelGate,
+)
+
+
+# pylint: disable=invalid-name
+
+
+# Basic layers
+class LightConvStream(nn.Module):
+    """Lightweight convolution stream."""
+
+    def __init__(self, in_channels: int, out_channels: int, depth: int) -> None:
+        super().__init__()
+        assert depth >= 1, f"depth must be equal to or larger than 1, but got {depth}"
+        layers = []
+        layers += [LightConv3x3(in_channels, out_channels)]
+        for _ in range(depth - 1):
+            layers += [LightConv3x3(out_channels, out_channels)]
+        self.layers = nn.Sequential(*layers)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Defines the computation performed at every call."""
+        return self.layers(x)
+
+
+# Building blocks for omni-scale feature learning
+class OSBlock(nn.Module):
+    """Omni-scale feature learning block."""
+
+    def __init__(  # pylint: disable=unused-argument
+        self,
+        in_channels: int,
+        out_channels: int,
+        reduction: int = 4,
+        T: int = 4,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__()
+        assert T >= 1
+        assert out_channels >= reduction and out_channels % reduction == 0
+        mid_channels = out_channels // reduction
+
+        self.conv1 = Conv1x1(in_channels, mid_channels)
+        self.conv2 = nn.ModuleList()
+        for t in range(1, T + 1):
+            self.conv2 += [LightConvStream(mid_channels, mid_channels, t)]
+        self.gate = ChannelGate(mid_channels)
+        self.conv3 = Conv1x1Linear(mid_channels, out_channels)
+        self.downsample = None
+        if in_channels != out_channels:
+            self.downsample = Conv1x1Linear(in_channels, out_channels)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Defines the computation performed at every call."""
+        identity = x
+        x1 = self.conv1(x)
+        x2 = 0
+        for conv2_t in self.conv2:
+            x2_t = conv2_t(x1)
+            x2 = x2 + self.gate(x2_t)
+        x3 = self.conv3(x2)
+        if self.downsample is not None:
+            identity = self.downsample(identity)
+        out = x3 + identity
+        return F.relu(out)
+
+
+class OSBlockINin(nn.Module):
+    """Omni-scale feature learning block with instance normalization."""
+
+    def __init__(  # pylint: disable=unused-argument
+        self,
+        in_channels: int,
+        out_channels: int,
+        reduction: int = 4,
+        T: int = 4,
```

Comment:
This shouldn't be all caps, also try to reference the paper if possible to use a self-descriptive name

Reply:
Changed to `num_streams`